tests/lib/rules/prefer-object-spread.js

```diff
@@ -0,0 +1,422 @@
+        "let a = Object.assign(a, b)",
+        "Object.assign(a, b)",
+        "let a = Object.assign(b, { c: 1 })",
+        "let a = Object.assign({}, ...b)",
+        "const bar = { ...foo }",
+        "Object.assign(...foo)",
+        "Object.assign(foo, { bar: baz })"
+    ],
+
+    invalid: [
+        {
+            code: "Object.assign({}, foo)",
+            output: "({...foo})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar' })",
+            output: "({foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, baz, { foo: 'bar' })",
+            output: "({...baz, foo: 'bar'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({}, { foo: 'bar', baz: 'foo' })",
+            output: "({foo: 'bar', baz: 'foo'})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code: "Object.assign({ foo: 'bar' }, baz)",
+            output: "({foo: 'bar', ...baz})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // Many args
+        {
+            code: "Object.assign({ foo: 'bar' }, cats, dogs, trees, birds)",
+            output: "({foo: 'bar', ...cats, ...dogs, ...trees, ...birds})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+
+        // TODO: Handle nested Object.assign calls
+        {
+            code:
+                "Object.assign({ foo: 'bar' }, Object.assign({ bar: 'foo' }, baz))",
+            output: "({foo: 'bar', ...Object.assign({ bar: 'foo' }, baz)})",
+            errors: [
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                },
+                {
+                    messageId: "useSpreadMessage",
+                    type: "CallExpression"
+                }
+            ]
+        },
+        {
+            code:
+                "Object.assign({ foo: 'bar' }, Object.assign({ bar: 'foo' }, Object.assign({}, { superNested: 'butwhy' })))",
+            output: "({foo: 'bar', ...Object.assign({ bar: 'foo' }, Object.assign({}, { superNested: 'butwhy' }))})",
```

Comment:
I think the innermost `Object.assign` should be identified and fixed-- is that a known limitation at this point? Or are we relying on multipass autofix to catch these cases? (Now the TODO above makes more sense...)

Reply:
Multipass, i think.

Reply:
I had a solution earlier that would recurse through nested object.assign calls, but the current implementation relies on multipass